### ISSUE TITLE
Implement __invert__ method on SArray

### ISSUE THREAD

**Issue by esamanas** [dato-code/SFrame#261]
This would only work if the type was `int`. It would allow a user to invert a boolean mask like so:

```
above3 = sf['rating'] > 3
lessorequal3 = ~above3
```

Right now, they have to do this:
```
above3 = sf['rating'] > 3
lessorequal3 = sf['rating'].apply(lambda x: 1 if x == 0 else 0)
```

**Comment by dnc1994** [comment]
I tried to implement `SArray.__invert__`, but run into some difficulties.

1. `~x` means taking the complement of `x`. If `x` is a boolean, then you get what you expected; however if `x` is a int, let's see how `numpy` implements this:
```
>>> import numpy as np
>>> mask = np.array([True, False])
>>> ~mask
array([False,  True], dtype=bool)
>>> int_mask = np.array([1, 0])
>>> ~int_mask
array([-2, -1])
```
In other words `~x = -x -1`.

The problem is that, in SArray, a boolean array is transformed into a int array.
```
>>> import graphlab as gl
>>> gl.SArray([True, False])
dtype: int
Rows: 2
[1L, 0L]
```
Therefore, implementing `~` for SArray demands that users have to check for themselves that the operand is indeed a boolean array.

2. According from some already implemented special methods in `SArray` class, we can do something like:
```
def __invert__(self):
    with cython_context():
        return SArray(_proxy = self.__proxy__.unary_operator('~'))
```
Sadly there is not `unary_operator`. We have to do it throught vector operator or scalar operator.
Then I thought we could have something like `mask ^= 1`. Then I found that the `^` is not supported in the underlying C++ codes. It seems from `oss_src/unity/lib/unity_sarray_binary_operations.cpp` that for now the supported operators are: `+, -, *, **, %, /, <, >, <=, >=, ==, !=, &, |, in, left_abs`.

I think the right way to do this is to dive into C++ codes and add support for more operators. It seems not too difficult, but as I'm not very experienced in C++, it could take a while for me to implement this.
Right now I've push a workaround (https://github.com/dato-code/SFrame/pull/263). It first checks whether `dtype` is int and whether it is a boolean array (by checking if it only contains 0 and 1).Then it returns `mask != 1`.

Folks I would like to know your thoughts on how to do this more properly.
